CSS module: logical props (#33901)
mdn/content: files/en-us/web/css/css_logical_properties_and_values/index.md

{{CSSRef}}

> The **CSS logical properties and values** module defines logical properties and values that provide the ability to control layout through logical, rather than physical, direction and dimension mappings Logical properties define direction‐relative equivalents of their corresponding physical properties.

Reviewer: ```suggestion
The **CSS logical properties and values** module defines logical properties and values that can control layout through logical rather than physical direction and dimension mappings. Logical properties define direction‐relative equivalents to their corresponding physical properties.
```

> The start of a line is not always the left side of a line. There are several different writing systems written in different directions. English and Portuguese are written from left to right, with new lines added below the previous ones. Hebrew and Arabic are right to left languages, also from top to bottom. In some writing modes, the text lines are vertical, written from top to bottom. Traditionally, Chinese, Vietnamese, Korean, and Japanese are written vertically, from top to bottom, with each new vertical line on the left of the previous one. Traditional Mongolian is also top to bottom, but new lines are to the right of previous ones. The logical properties defined in this module enable defining properties relative to the content's writing directions, rather than a physical direction meaning content translated into languages with different writing modes will render as intended.

Reviewer: ```suggestion
The start of a line is not always the left side of a line. Different writing systems operate in various directions. For example:

- English and Portuguese are written from left to right with new lines added below the previous ones.
- Hebrew and Arabic are right-to-left languages with new lines again being added below the previous ones.
- In some writing modes, the text lines are vertical, written from top to bottom. Chinese, Vietnamese, Korean, and Japanese are traditionally written vertically, from top to bottom, with each new vertical line added to the left of the previous one.
- Traditional Mongolian is also a top-to-bottom language, but new lines are to the right of previous ones.

The logical properties defined in this module enable defining properties relative to the content's writing direction, rather than a physical direction. This means content translated into languages with different writing modes will be rendered as intended.
```

Logical properties and values use the abstract terms _block_ and _inline_ to describe the direction in which they flow. The physical meaning of these terms depends on the [writing mode](/en-US/docs/Web/CSS/CSS_writing_modes).

The **block dimension** is perpendicular to the flow of text within a line, i.e., the vertical dimension in horizontal writing modes, and the horizontal dimension in vertical writing modes. For standard English text, it is the vertical dimension.

The **inline dimension** is parallel to the flow of text within a line, i.e., the horizontal dimension in horizontal writing modes, and the vertical dimension in vertical writing modes. For standard English text, it is the horizontal dimension.

> CSS was initially designed with only physical coordinates in its controls. The logical properties and values module defines flow–relative equivalents for many [values](/en-US/docs/Web/CSS/CSS_Values_and_Units) and [properties](/en-US/docs/Glossary/Property/CSS). Properties that once only accepted physical values (`top`, `bottom`, `left`, `right`) also accept flow-relative logical values (`block-start`, `block-end`, `inline-start`, `inline-end`).

Reviewer: ```suggestion
CSS was initially designed with only physical coordinates. The logical properties and values module defines flow–relative equivalents for many [values](/en-US/docs/Web/CSS/CSS_Values_and_Units) and [properties](/en-US/docs/Glossary/Property/CSS). Properties that once only accepted physical values (`top`, `bottom`, `left`, `right`) now also accept flow-relative logical values (`block-start`, `block-end`, `inline-start`, `inline-end`).
```

- {{cssxref("margin-block")}}
- {{cssxref("margin-block-end")}}
- {{cssxref("margin-block-start")}}
- {{cssxref("margin-inline")}}
- {{cssxref("margin-inline-end")}}
- {{cssxref("margin-inline-start")}}
- {{cssxref("max-block-size")}}
- {{cssxref("max-inline-size")}}
- {{cssxref("min-block-size")}}
- {{cssxref("min-inline-size")}}
- {{cssxref("padding-block")}}
- {{cssxref("padding-block-end")}}
- {{cssxref("padding-block-start")}}
- {{cssxref("padding-inline")}}
- {{cssxref("padding-inline-end")}}
- {{cssxref("padding-inline-start")}}

### Data types and values

{{glossary("Flow relative values")}}:

- `block-start`
- `block-end`
- `inline-start`
- `inline-end`
- `start`
- `end`

### Glossary terms

- {{glossary("Flow relative values")}}
- {{glossary("Inset properties")}}
- {{glossary("Logical properties")}}
- {{glossary("Physical properties")}}

## Guides

- [Basic concepts of logical properties and values](/en-US/docs/Web/CSS/CSS_logical_properties_and_values/Basic_concepts_of_logical_properties_and_values)

  - : Overview of flow relative properties and values.

- [Logical properties for sizing](/en-US/docs/Web/CSS/CSS_logical_properties_and_values/Sizing)

>   - : Flow-relative mappings between physical dimension properties and logical properties used for sizing elements on the page

Reviewer: ```suggestion
  - : Flow-relative mappings between physical properties and logical properties used for sizing elements on the page.
```

- [Logical properties for margins, borders, and padding](/en-US/docs/Web/CSS/CSS_logical_properties_and_values/Margins_borders_padding)

  - : Flow-relative mappings for the various margin, border, and padding properties and their shorthands.

- [Logical properties for floating and positioning](/en-US/docs/Web/CSS/CSS_logical_properties_and_values/Floating_and_positioning)

>   - : The logical mappings for the physical values for `float` and `clear`, for the positioning properties used with positioned layout, and for `resize`.

Reviewer: ```suggestion
  - : Details mappings between the physical and logical values for `float` and `clear`, [inset properties](/en-US/docs/Glossary/Inset_properties), and  `resize`.
```
